src/Framework/ContaoFramework.php

```diff
@@ -476,4 +477,34 @@ private function canSkipTokenCheck(): bool
             || false === $this->request->attributes->get('_token_check')
         ;
     }
+
+    /**
+     * Build hooks globals.
+     */
+    private function buildHookGlobals(): void
+    {
+        if ($this->container->hasParameter('contao.hook_listeners')) {
+            $config = (array) $this->container->getParameter('contao.hook_listeners');
+
+            foreach ($config as $hookName => $priorities) {
+                $hooks = [];
+
+                foreach ($priorities as $priority => $listeners) {
```

Comment:
I think this can be simplified to something like this:

```php
if (isset($GLOBALS['TL_HOOKS'][$hookName])) {
    if (isset($priorities[0]) {
        $priorities[0] = array_merge($GLOBALS['TL_HOOKS'][$hookName], $priorities[0]);
    } else {
        $priorities[0] = $GLOBALS['TL_HOOKS'][$hookName];
    }
}

$GLOBALS['TL_HOOKS'][$hookName] = array_merge_recursive($priorities);
```

Reply:
No, because array_merge_recursive would merge like this:

```php
$arrayA = ['foo','bar'];
$arrayB = ['baz'];

// would create [['foo', 'baz'], 'bar]
$merged = array_merge_recursive($arrayA, $arrayB);
```